### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by iakhator**
* For peer review

### CODE REVIEW

File: coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rhenshaw56) [inline coverage/clover.xml]: I don't think you're supposed to put your coverage folder on github

File: src/components/App.jsx [hunk 2]
```diff
@@ -38,16 +36,14 @@ export default class App extends React.Component {
 			isAuthenticated: AuthStore.isAuthenticated(),
 			displayName: ''
```
Comment (rhenshaw56) [inline src/components/App.jsx]: if you're not going to use `props` anywhere inside the constructor whats the need of passing it to `super()` as `super(props)`.
Just doing `super()` without `props` is fine

File: src/components/NewsFeeds.jsx [hunk 3]
```diff
@@ -4,24 +4,46 @@ import { Link } from 'react-router-dom'
 import NewsActions from '../actions/NewsActions';
 import NewsStore from '../stores/NewsStore';
 import AuthStore from '../stores/AuthStore';
-import AuthActions from '../actions/AuthActions';
 
+/**
```
Comment (rhenshaw56) [inline src/components/NewsFeeds.jsx]: your `jsDocs` does not tell us what this function does. No description

File: src/components/NewsFeeds.jsx [hunk 4]
```diff
@@ -4,24 +4,46 @@ import { Link } from 'react-router-dom'
...
+ *
+ * @param {*} props
```
Comment (rhenshaw56) [inline src/components/NewsFeeds.jsx]: is your `eslint` on ?

File: src/components/NewsHeadline.jsx [hunk 5]
```diff
@@ -40,34 +40,34 @@ function Headline(props) {
...
 		this.state = {
-			newsSource: null
+			newsArticle: null
 		};
 		this.recieveSource = this.recieveSource.bind(this);
 	}
 
 	componentDidMount() {
-		NewsActions.getSource(this.props.match.params.sourceId);
+		NewsActions.getArticle(this.props.match.params.sourceId, this.props.match.params.sortBy);
```
Comment (rhenshaw56) [inline src/components/NewsHeadline.jsx]: Please check up on your `eslint`

File: src/components/header/Nav.jsx [hunk 6]
```diff
@@ -1,57 +1,77 @@
...
-								<NavLink activeClassName="active" to="/newsfeeds">
-									NewsFeeds
-							</NavLink>
-							</li>
-						}
-					</ul>
-					<ul className="nav navbar-nav navbar-right">
-						{authenticate &&
-							<ul className="nav navbar-nav navbar-right">
+export default class Nav extends React.Component {
```
Comment (rhenshaw56) [inline src/components/header/Nav.jsx]: I'm sure you've refactored it [here](https://github.com/iakhator/news-feed-application/blob/develop/src/components/header/Nav.jsx#L16) but i would suggest you using a vs-code jsDoc plugin called `Document this`

File: src/helpers/auth.js [hunk 7]
```diff
@@ -2,13 +2,16 @@ import { firebaseAuth, provider } from '../config/firebase-config';
...
+
+	auth: () => {
+		return firebaseAuth.getRedirectResult().then(function(result) {
 			return result;
-		}).catch((error) => {
-			console.log(error.code);
+		}).catch(function(error) {
+			console.log(error)
 		});
+
```
Comment (rhenshaw56) [inline src/helpers/auth.js]: i see that after creating the class, you instantiated it and then exported the instance [here](https://github.com/iakhator/news-feed-application/blob/develop/src/helpers/auth.js#L18-L19). But why?

File: .codeclimate.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+engines:
+ duplication:
+   enabled: false
+   config:
+     languages:
+       javascript:
+         mass_threshold: 20
+         exclude_paths:
+           - “src/__tests__/”
```
Comment (andela-tisrael) [inline .codeclimate.yml]: New line after this

File: .eslintignore [hunk 9]
```diff
@@ -0,0 +1,4 @@
+node_modules/
+src/__tests__/
+public/
+webpack.config.js
```
Comment (andela-tisrael) [inline .eslintignore]: New line after this

File: .eslintrc.js [hunk 10]
```diff
@@ -1,32 +1,295 @@
...
+        "never"
+      ],
+      "valid-jsdoc": "off",
+      "vars-on-top": "off",
+      "wrap-iife": "off",
+      "wrap-regex": "off",
+      "yield-star-spacing": "error",
+      "yoda": "off"
+    }
 };
```
Comment (andela-tisrael) [inline .eslintrc.js]: New line after this

File: .gitignore [hunk 11]
```diff
@@ -1 +1,6 @@
-/node_modules
\ No newline at end of file
+node_modules
+coverage
+.nyc_output
+.env
+.DS_Store
+.lcov.info
```
Comment (andela-tisrael) [inline .gitignore]: New line after this

File: .hound.yml [hunk 12]
```diff
@@ -0,0 +1,4 @@
+eslint:
+  enabled: true
+  config_file: .eslintrc
+  ignore_file: .eslintignore
```
Comment (andela-tisrael) [inline .hound.yml]: New line after this

File: .travis.yml [hunk 13]
```diff
@@ -1,3 +1,4 @@
 language: node_js		  
 node_js: "stable"
-script: npm run test
\ No newline at end of file
+test: "npm test --coverage"
+after_success: "cat ./coverage/lcov.info | coveralls"
```
Comment (andela-tisrael) [inline .travis.yml]: New line after this

File: Procfile [hunk 14]
```diff
@@ -0,0 +1 @@
+web: npm start
```
Comment (andela-tisrael) [inline Procfile]: New line after this

File: public/index.html [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+	<title>News Feeds</title>
+	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+<link href="/css/index.bundle.css" rel="stylesheet"></head>
+<body>
+	<div id="app"></div>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js"></script>
+	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+<script type="text/javascript" src="/bundle.js"></script></body>
+</html>
```
Comment (andela-tisrael) [inline public/index.html]: New line after this

File: server.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+
+const express = require('express');
+const path = require('path');
+
+const app = express();
+
+//app.set('port', '3000');
```
Comment (andela-tisrael) [inline server.js]: Remove commented code

File: server.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+app.use(express.static(path.join(__dirname, 'public')));
+
+// app.get('*', function (request, response){
```
Comment (andela-tisrael) [inline server.js]: Remove commented code

File: src/__tests__/action/NewsActions.test.jsx [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+    }); 
+    
+    it('should dispatch an action', () => 
+      NewsActions.getArticle('abc-news-au', 'top')
+        .then(() => {
+          const dispatcherCall = spyOnDispatcher.mock.calls[0][0];
+          expect(spyOnDispatcher).toHaveBeenCalled();
+          expect(dispatcherCall.type).toEqual('RECIEVE_ARTICLE');
+          expect(dispatcherCall.article).toBeInstanceOf(Object);
+          //expect(dispatcherCall.article).toEqual(articles);
```
Comment (andela-tisrael) [inline src/__tests__/action/NewsActions.test.jsx]: Remove commented code

File: src/__tests__/action/NewsActions.test.jsx [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+        })
+    );
+
+    
+  });
+  
+});
```
Comment (andela-tisrael) [inline src/__tests__/action/NewsActions.test.jsx]: New line after this

File: src/components/Headline.jsx [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+      })}
+    </div>
+  );
+}
+
+Headline.PropTypes = {
+  newsArticle: PropTypes.array.isRequired,
+};
+
+export default Headline;
```
Comment (andela-tisrael) [inline src/components/Headline.jsx]: New Line after this

File: src/components/NewsFeeds.jsx [hunk 21]
```diff
@@ -1,108 +1,95 @@
...
-			</div>
-		);
-	}
+  render() {
+    return (
+      <div className="body">
+        <div className="container contain">
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <p className="para text-center ">
```
Comment (andela-tisrael) [inline src/components/NewsFeeds.jsx]: I don't think `para` explains what the html block is actually doing

File: src/components/NewsFeeds.jsx [hunk 22]
```diff
@@ -1,108 +1,95 @@
...
+        </div>
+      </div>
+    );
+  }
 }
 
+News.PropTypes = {
 
+}
+export default News;
```
Comment (andela-tisrael) [inline src/components/NewsFeeds.jsx]: New Line after this

File: src/components/NewsHeadline.jsx [hunk 23]
```diff
@@ -1,89 +1,63 @@
...
-							</div>
-						</div>
-					</div>
-				</div>
-			</div>
-		);
-	}
-}
\ No newline at end of file
+export default NewsHeadline;
```
Comment (andela-tisrael) [inline src/components/NewsHeadline.jsx]: New Line after this

File: src/components/Sources.jsx [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+    </div>
+  );
+}
+
+Sources.PropTypes = {
+  newsSource: PropTypes.array.isRequired,
+  search: PropTypes.string.isRequired
+}
+
+export default Sources;
```
Comment (andela-tisrael) [inline src/components/Sources.jsx]: New Line after this

File: src/components/header/Nav.jsx [hunk 25]
```diff
@@ -1,57 +1,104 @@
...
+   * The logout method initialize the signing out of
+   * the user and also set the state
+   */
+  logOut() {
+    if (this.state.isAuthenticated) {
+      Auth.logOut();
+      this.setState({
+        isAuthenticated: false
+      })
+      //history.push('/');
```
Comment (andela-tisrael) [inline src/components/header/Nav.jsx]: Remove commented code

File: src/helpers/auth.js [hunk 26]
```diff
@@ -1,22 +1,19 @@
...
-		})
+	logOut() {
+		return firebaseAuth.signOut();
 	}
-}
\ No newline at end of file
+}
+
+const auth = new Auth()
+export default auth;
```
Comment (andela-tisrael) [inline src/helpers/auth.js]: New Line

File: src/sass/index.scss [hunk 27]
```diff
@@ -99,8 +101,10 @@ button{
 
 .navbar-default{
 	 border: none;
-	 padding: 12px;
    .navbar-nav li{
+		 a{
```
Comment (andela-tisrael) [inline src/sass/index.scss]: You might want to look into spacing in this css file, It looks like 8 spaces, but 2 or 4 spaces is okay

File: src/sass/index.scss [hunk 28]
```diff
@@ -128,16 +132,51 @@ button{
 }
 
 .newsfeed{
-	display: flex !important;
-	flex-wrap: wrap;
+	// display: flex !important;
```
Comment (andela-tisrael) [inline src/sass/index.scss]: Remove the commented css property

**Pull request merged**